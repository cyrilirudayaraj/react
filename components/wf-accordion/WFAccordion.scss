@import '~@athena/forge/sass/forge-abstracts';

.wf-accordion {
  .fe_c_accordion-item__content {
    .fe_c_accordion-item__padded-content {
      padding-left: 14px !important;
    }
  }
  .wf-accordion-item .fe_c_accordion-item__expand {
    position: initial !important;
  }
  .wf-accordion-summary {
    width: 100%;
    margin-left: 10px;
    .fe_c_heading--subsection {
      font-size: 16px;
    }
  }

  .fe_c_accordion-item__header {
    background-color: white !important;
  }

  .fe_is-expanded {
    border: none !important;
    box-shadow: none !important;
  }
}
