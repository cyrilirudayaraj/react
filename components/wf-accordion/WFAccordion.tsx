import { Accordion, AccordionProps } from '@athena/forge';
import React from 'react';

const WFAccordion = React.forwardRef(function WFAccordion(
  props: AccordionProps,
  ref: any
) {
  const className = props.className || '';
  return <Accordion {...props} className={`wf-accordion ${className}`} />;
});

export default WFAccordion;
