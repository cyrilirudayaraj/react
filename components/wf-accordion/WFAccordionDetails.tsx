import React from 'react';

const WFAccordionDetails = React.forwardRef(function WFAccordionDetails(
  props: any,
  ref: any
) {
  return (
    <div className="wf-accordion-details fe_c_accordion-item__content">
      {props.children}
    </div>
  );
});
export default WFAccordionDetails;
