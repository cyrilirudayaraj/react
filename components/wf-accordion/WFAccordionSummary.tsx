import React from 'react';

export interface WFAccordionSummaryProps {
  expanded: boolean;
  exited: boolean;
  toggleExpand: any;
}
const WFAccordionSummary = React.forwardRef(function WFAccordionSummary(
  props: any,
  ref: any
) {
  return <div className="wf-accordion-summary">{props.children}</div>;
});

export default WFAccordionSummary;
