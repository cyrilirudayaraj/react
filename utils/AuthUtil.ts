const winObj: any = window;

class AuthUtil {
  static getLoggedInUsername(): string {
    return winObj.globalVars
      ? winObj.globalVars.username
      : process.env.REACT_APP_USER;
  }

  static isPermissionEnabled(): boolean {
    const enablePermission = winObj.globalVars
      ? winObj.globalVars.enablePermission
      : process.env.REACT_APP_ENABLE_PERMISSION;
    return enablePermission && enablePermission == 1;
  }

  static isAuthorized(
    permission: string,
    userPermissionList: string[]
  ): boolean {
    let result = true;
    const eventList = userPermissionList ? userPermissionList : [];
    if (AuthUtil.isPermissionEnabled()) {
      if (permission && !eventList.includes(permission)) {
        result = false;
      }
    }
    return result;
  }

  static isConsecutiveSignoffDisabled(): boolean {
    const disableConsecutiveSignoff = winObj.globalVars
      ? winObj.globalVars.disableConsecutiveSignoff
      : process.env.REACT_APP_DISABLE_CONSECUTIVE_SIGNOFF;
    return disableConsecutiveSignoff && disableConsecutiveSignoff == 1;
  }

  static disableConsecutiveSignoff(previousSignoffUsername: string): boolean {
    if (
      AuthUtil.isConsecutiveSignoffDisabled() &&
      AuthUtil.getLoggedInUsername() === previousSignoffUsername
    ) {
      return true;
    }
    return false;
  }
}

export default AuthUtil;
