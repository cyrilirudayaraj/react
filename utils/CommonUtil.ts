import moment from 'moment';
import AppConstants from '../constants/AppConstants';
import { TaskStep } from '../types';

class CommonUtil {
  static handleDateChange(event: any, formik: any): void {
    const { value } = event.target;
    let output: any = undefined;
    if (value) {
      const momentDate = moment(
        value,
        AppConstants.UI_CONSTANTS.DATE_INPUT_FORMAT,
        true
      );
      if (momentDate.isValid()) {
        output = momentDate.toDate();
      }
    }
    formik.setFieldValue(event.target.id, output, false);
  }
  static getPreviousTaskStep(taskSteps: any, activeTaskStep: any): any {
    const previousStep = taskSteps.filter(
      (taskStep: TaskStep) =>
        parseInt(taskStep.ordering) === parseInt(activeTaskStep.ordering) - 1
    );
    return previousStep[0];
  }

  static isActiveTaskType(taskTypeId: string | undefined): boolean {
    if (
      taskTypeId ===
        AppConstants.SERVER_CONSTANTS.TASK_TYPES.BUSINESS_REQUIREMENT_UPDATE ||
      taskTypeId ===
        AppConstants.SERVER_CONSTANTS.TASK_TYPES.NEW_BUSINESS_REQUIREMENT ||
      taskTypeId ===
        AppConstants.SERVER_CONSTANTS.TASK_TYPES.DUAL_MAINTENANCE_BR_UPDATE ||
      taskTypeId ===
        AppConstants.SERVER_CONSTANTS.TASK_TYPES.DUAL_MAINTENANCE_NEW_BR
    ) {
      return true;
    }
    return false;
  }
}
export default CommonUtil;
