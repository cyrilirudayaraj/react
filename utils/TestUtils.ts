//TODO exclude this file during bundling
export function findByName(wrapper: any, name: string): any {
  const selector = "input[name='" + name + "']";
  return wrapper.find(selector).last();
}

export function findById(wrapper: any, id: string): any {
  const selector = '#' + id;
  return wrapper.find(selector).last();
}

export function changeValueById(wrapper: any, id: string, value: string): void {
  findById(wrapper, id).simulate('change', {
    target: { name: id, value: value },
  });
}

export function sleep(x: number): Promise<any> {
  return new Promise<void>((resolve) => {
    setTimeout(() => {
      resolve();
    }, x);
  });
}
