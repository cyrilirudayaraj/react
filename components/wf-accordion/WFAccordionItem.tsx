import { Icon, ListItem } from '@athena/forge';
import React, { Component } from 'react';
import { CSSTransition } from 'react-transition-group';
import StringUtil from '../../utils/StringUtil';
import './WFAccordion.scss';

const classNames = StringUtil.classNames;

interface WFAccordionItemState {
  expanded: boolean | undefined;
  exited: boolean;
}

interface WFAccordionItemProps {
  className?: string;
  defaultExpanded?: boolean;
  mountedWhileHidden?: boolean;
  expanded?: boolean;
  padded?: boolean;
  onExpandedChange?: (...args: any[]) => any;
}

export default class WFAccordionItem extends Component<
  WFAccordionItemProps,
  WFAccordionItemState
> {
  constructor(props: WFAccordionItemProps) {
    super(props);

    const expanded =
      props.expanded !== undefined ? props.expanded : props.defaultExpanded;
    this.state = {
      expanded: expanded,
      exited: !expanded,
    };
  }

  static getDerivedStateFromProps(newProps: WFAccordionItemProps): any {
    if (WFAccordionItem.inControlledMode(newProps)) {
      return {
        expanded: newProps.expanded,
      };
    }
    return null;
  }

  static inControlledMode(props: WFAccordionItemProps): boolean {
    return props.expanded !== undefined;
  }

  toggleExpand = () => {
    this.setState((prevState: any, props: any) => ({
      expanded: WFAccordionItem.inControlledMode(props)
        ? props.expanded
        : !prevState.expanded,
    }));

    if (this.props.onExpandedChange) {
      this.props.onExpandedChange(!this.state.expanded);
    }
  };

  handleEnter = (content: HTMLElement): void => {
    this.setState({ exited: false });
    content.style.height = '0';
  };

  handleEntering = (content: HTMLElement): void => {
    content.style.height = `${content.scrollHeight}px`;
  };

  handleEntered = (content: any): void => {
    content.style.height = null;
  };

  handleExit = (content: HTMLElement): void => {
    content.style.height = `${content.scrollHeight}px`;
  };

  handleExiting = (content: HTMLElement): void => {
    content.style.height = '0';
  };

  handleExited = (content: HTMLElement): void => {
    this.setState({ exited: true });
  };

  render(): JSX.Element {
    const {
      mountedWhileHidden,
      children: childrenProp,
      padded,
      ...passedProps
    } = this.props;

    const { exited } = this.state;

    const [summary, ...detailchildren] = React.Children.toArray(childrenProp);

    return (
      <ListItem
        className={classNames(
          'fe_c_list__item',
          'fe_c_accordion-item',
          'wf-accordion-item',
          {
            'fe_is-expanded': this.state.expanded,
          }
        )}
        {...passedProps}
      >
        <button
          className={classNames('fe_c_accordion-item__header', {
            'fe_is-expanded': this.state.expanded,
          })}
          onClick={this.toggleExpand}
          aria-expanded={!exited}
          type="button"
        >
          {summary}

          <Icon
            icon="Expand"
            title={this.state.expanded ? 'Collapse' : 'Expand'}
            aria-hidden="true"
            focusable="false"
            className={classNames('fe_c_accordion-item__expand', {
              'fe_is-expanded': this.state.expanded,
            })}
          />
        </button>
        <div
          className={classNames('fe_c_accordion-item__content', {
            'fe_is-expanded': this.state.expanded,
          })}
        >
          <CSSTransition
            in={this.state.expanded}
            classNames="fe_is-expanding"
            timeout={400}
            onEnter={this.handleEnter}
            onEntering={this.handleEntering}
            onEntered={this.handleEntered}
            onExit={this.handleExit}
            onExiting={this.handleExiting}
            onExited={this.handleExited}
            mountOnEnter={!mountedWhileHidden}
            unmountOnExit={!mountedWhileHidden}
          >
            <div className="fe_c_accordion-item__content-inner">
              {padded ? (
                <div className="fe_c_accordion-item__padded-content">
                  {detailchildren}
                </div>
              ) : (
                detailchildren
              )}
            </div>
          </CSSTransition>
        </div>
      </ListItem>
    );
  }
}
