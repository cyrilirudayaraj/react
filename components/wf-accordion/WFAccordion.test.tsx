import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WFAccordion from './WFAccordion';
import WFAccordionItem from './WFAccordionItem';
import WFAccordionDetails from './WFAccordionDetails';
import WFAccordionSummary from './WFAccordionSummary';
import { Heading } from '@athena/forge';
import { sleep } from '../../utils/TestUtils';
import { act } from 'react-dom/test-utils';

configure({ adapter: new Adapter() });
let wrapper: any;

beforeEach(() => {
  wrapper = mount(
    <WFAccordion>
      <WFAccordionItem padded={true}>
        <WFAccordionSummary>
          <div>
            <Heading headingTag="h5" variant="subsection" text="Header here" />
            <p>Keep Here the minimum details</p>
          </div>
        </WFAccordionSummary>
        <WFAccordionDetails>
          <div>
            Lorem Ipsum is simply dummy text of the printing and typesetting
            industry. Lorem Ipsum has been the industry's standard dummy text
            ever since the 1500s, when an unknown printer took a galley of type
            and scrambled it to make a type specimen book.{' '}
          </div>
        </WFAccordionDetails>
      </WFAccordionItem>
    </WFAccordion>
  );
});

describe('snapshot test <WFSelect>', () => {
  it('should render as expected', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should expand when Accordion summary clicked', () => {
    wrapper.find(WFAccordionSummary).first().simulate('click');

    expect(wrapper.find('.fe_c_accordion-item__content-inner').length).toEqual(
      1
    );
  });

  it('should collapsed when Accordion summary clicked twice', async () => {
    const summaryEl = wrapper
      .find('button.fe_c_accordion-item__header')
      .first();
    await act(async () => {
      await sleep(1);

      summaryEl.simulate('click');
      await sleep(1);
      expect(
        wrapper
          .find('.fe_c_accordion-item__content')
          .first()
          .is('.fe_is-expanded')
      ).toEqual(true);

      summaryEl.simulate('click');
      await sleep(1);
      expect(
        wrapper
          .find('.fe_c_accordion-item__content')
          .first()
          .is('.fe_is-expanded')
      ).toEqual(false);
    });
  });
});
