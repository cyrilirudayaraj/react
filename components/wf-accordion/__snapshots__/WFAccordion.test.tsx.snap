// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test <WFSelect> should render as expected 1`] = `
"<ForwardRef(WFAccordion)>
  <Accordion className=\\"wf-accordion \\">
    <List className=\\"fe_c_accordion wf-accordion\\" padded={false} dividers={false}>
      <ul className=\\"fe_c_list fe_c_accordion wf-accordion\\">
        <WFAccordionItem padded={true}>
          <ListItem className=\\"fe_c_list__item fe_c_accordion-item wf-accordion-item\\">
            <li className=\\"fe_c_list__item fe_c_list__item fe_c_accordion-item wf-accordion-item\\">
              <button className=\\"fe_c_accordion-item__header\\" onClick={[Function (anonymous)]} aria-expanded={false} type=\\"button\\">
                <ForwardRef(WFAccordionSummary)>
                  <div className=\\"wf-accordion-summary\\">
                    <div>
                      <Heading headingTag=\\"h5\\" variant=\\"subsection\\" text=\\"Header here\\">
                        <h5 className=\\"fe_c_heading fe_c_heading--subsection\\">
                          Header here
                        </h5>
                      </Heading>
                      <p>
                        Keep Here the minimum details
                      </p>
                    </div>
                  </div>
                </ForwardRef(WFAccordionSummary)>
                <Icon icon=\\"Expand\\" title=\\"Expand\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_accordion-item__expand\\" iconSet={{...}}>
                  <Expand title=\\"Expand\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_accordion-item__expand\\">
                    <IconBase viewBox=\\"0 0 25 25\\" title=\\"Expand\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_accordion-item__expand\\">
                      <svg xmlns=\\"http://www.w3.org/2000/svg\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_icon fe_c_accordion-item__expand\\" viewBox=\\"0 0 25 25\\" style={{...}}>
                        <title>
                          Expand
                        </title>
                        <g>
                          <g className=\\"fe_c_icon__background\\">
                            <rect width=\\"25\\" height=\\"25\\" />
                          </g>
                          <g className=\\"fe_c_icon__icon fe_c_icon__icon--primary\\">
                            <polygon points=\\"12.5 16.6 6 10 7 9 12.5 14.4 18 9 19 10 12.5 16.6\\" />
                          </g>
                        </g>
                      </svg>
                    </IconBase>
                  </Expand>
                </Icon>
              </button>
              <div className=\\"fe_c_accordion-item__content\\">
                <CSSTransition in={[undefined]} classNames=\\"fe_is-expanding\\" timeout={400} onEnter={[Function (anonymous)]} onEntering={[Function (anonymous)]} onEntered={[Function (anonymous)]} onExit={[Function (anonymous)]} onExiting={[Function (anonymous)]} onExited={[Function (anonymous)]} mountOnEnter={true} unmountOnExit={true}>
                  <Transition in={false} timeout={400} onEnter={[Function (anonymous)]} onEntering={[Function (anonymous)]} onEntered={[Function (anonymous)]} onExit={[Function (anonymous)]} onExiting={[Function (anonymous)]} onExited={[Function (anonymous)]} mountOnEnter={true} unmountOnExit={true} appear={false} enter={true} exit={true} />
                </CSSTransition>
              </div>
            </li>
          </ListItem>
        </WFAccordionItem>
      </ul>
    </List>
  </Accordion>
</ForwardRef(WFAccordion)>"
`;
